docgen: add alter virtual cluster diagram

katmayb opened the pull request:
Epic: none

Release note: None

Release justification: non-production code change

Commit:
docgen: add alter virtual cluster diagram

Epic: none

Release note: None

Release justification: non-production code change

katmayb commented:
<img width="1040" alt="Screenshot 2025-05-08 at 2 13 11 PM" src="https://github.com/user-attachments/assets/2b2e728c-9618-43b1-b3e1-b03d080493e4" />
These included changes would result in the attached image.

taroface commented on pkg/cmd/docgen/diagrams.go:
```diff
@@ -575,6 +575,29 @@ var specs = []stmtSpec{
+		replace: map[string]string{"'ON' d_expr": "'ON' physical_cluster", "d_expr": "virtual_cluster_spec", "a_expr": "timestamp"},
+		unlink:  []string{"physical_cluster", "virtual_cluster_spec", "timestamp"},
+	},
+	{
```
I'm unclear on why the following 4 entries are necessary, unless you intend to use the separate generated BNF files elsewhere. If you inlined them in the `alter_virtual_cluster` definition, shouldn't they all appear under that single diagram (as you showed in the screenshot)?

katmayb replied:
hmmmm, it seemed like they were necessary when I used `dev generate bnf`... that is, I followed the error messages and added the required stmt entries. I can take another look if you've seen that these aren't necessary.

taroface replied:
If you followed the output then I think that's fine!

katmayb replied:
Just to note here as well, I followed step 4 and the content looked good 👍 Thanks!

taroface reviewed (commented):
Just 1 comment/question. Also, I wanted to check if you followed the step (in [step 4](https://cockroachlabs.atlassian.net/wiki/spaces/ED/pages/1134166645/SQL+Grammar+Documentation#Update-diagram-source-and-generate-diagram)) to verify that the actual HTML files have the updated content that you want. If not, I recommend doing this, since this is a more involved change.

katmayb commented:
bors r=msbutler

katmayb commented:
bors r=msbutler

Commit:
docgen: add alter virtual cluster diagram

Epic: none

Release note: None
Release justification: non-production code change

katmayb commented:
bors r-

katmayb commented:
bors r+
